### ISSUE TITLE
Slowness in EmberPerf

### ISSUE THREAD

**Issue by jonathandturner** [servo/servo#17329]
I did a comparison of EmberPerf's performance between Servo and the most recent stable Firefox, Safari, and Chrome.

You can see that generally Servo is the slowest of the bunch, though there are a couple places that Servo does decently.

![screen shot 2017-06-12 at 14 02 14](https://user-images.githubusercontent.com/547158/27158677-46ba6750-51bc-11e7-8ff9-ad97a5b2a7a0.png)

**Comment by emilio** [comment]
JSC people are gross.

**Comment by emilio** [comment]
JSC people are gross.

**Comment by emilio** [comment]
Getting profiles for this could be useful. For example, @glennw filed #17308 this morning, and we already found a bunch of dumb stuff going on (#17315)

**Comment by jonathandturner** [comment]
@emilio - I'd be happy to help.  What's the best way to get a profile?

**Comment by jonathandturner** [comment]
@emilio - I don't understand why you're calling people gross.

**Comment by emilio** [comment]
Sorry, my fault, not a native english speaker and got it wrong. Meant it in the sense that they're awesome JIT compiler writers, thanks for the correction. I've edited the comment.

**Comment by emilio** [comment]
> @emilio - I'd be happy to help. What's the best way to get a profile?

`perf record` works fine on Linux, we have a `-p` option to set a profile interval, but I think @glennw has reported before it may be lying some times? So perf (or instruments on OSX, maybe?) may be good tools.

I can also take a look if you tell me a good way to get STR... Though I guess it's just loading and running http://emberperf.eviltrout.com/... I can get some profiles.

**Comment by jonathandturner** [comment]
Yeah, to get it working I just tried running http://emberperf.eviltrout.com/ through all the default tests.  Sometimes it takes a couple tries since you might get an error that something isn't defined sometimes.

**Comment by jonathandturner** [comment]
I'll look into instruments on macOS and see what I can see

**Comment by emilio** [comment]
From a quick look with `perf record`, it's relatively hard to get some clear numbers on what's slow, apart from the JS engine. The fact that we're using a really old SpiderMonkey doesn't help either... :(

I've seen `Stylist::set_device` and `RuleIterator::next` appear in the profile, going to take a look at them. The fact that the style locks still appear around doesn't make me super happy (cc @SimonSapin), but I guess it's fine.
